### PR TITLE
[FLINK-24648] CEIL and FLOOR should support DECADE, CENTURY, MILLENNIUM

### PR DESCRIPTION
**Pull request by snuyanzin**
## What is the purpose of the change

The PR fixes return results for quieries like

```sql
select ceil(current_timestamp to decade);
select ceil(current_timestamp to decade);
select ceil(current_timestamp to decade);
```

## Verifying this change
  The tests are at `flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/TemporalTypesTest.scala`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by snuyanzin** [comment]
@tsreaper thanks, done

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/TemporalTypesTest.scala [hunk 1]
```diff
@@ -890,14 +890,24 @@ class TemporalTypesTest extends ExpressionTestBase {
     testSqlApi("FLOOR( DATE '2021-03-01' TO WEEK)", "2021-02-28")
     testSqlApi("CEIL( DATE '2021-02-27' TO WEEK)", "2021-02-28")
     testSqlApi("CEIL( DATE '2021-03-01' TO WEEK)", "2021-03-07")
-
+    testSqlApi("CEIL(DATE '2018-01-02' TO DECADE)", "2020-01-01")
+    testSqlApi("CEIL(DATE '2018-03-27' TO CENTURY)", "2101-01-01")
```
Comment (JingsongLi) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/TemporalTypesTest.scala:894]: It really surprised me... not "2100-01-01"....

**Review by JingsongLi — APPROVED** [review]
Thanks @snuyanzin , looks good to me.

**Comment by JingsongLi** [comment]
Hi @snuyanzin , I'm a little confused. Is the beginning of the era 00 or 01? Have you investigated the behavior of other databases?

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/TemporalTypesTest.scala (hunk 1 above)
Comment (tsreaper) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/TemporalTypesTest.scala:894]: The expected results are from postgre and vertica. See description of FLINK-24648.

From document of postgre:
> The first century starts at 0001-01-01 00:00:00 AD, although they did not know it at the time. This definition applies to all Gregorian calendar countries. There is no century number 0, you go from -1 century to 1 century. If you disagree with this, please write your complaint to: Pope, Cathedral Saint-Peter of Roma, Vatican.

**Pull request merged**